packages/pyright-internal/src/parser/tokenizer.ts

```diff
@@ -1572,17 +1569,22 @@ export class Tokenizer {
         const isTriplicate = (flags & StringTokenFlags.Triplicate) !== 0;
         const isFString = (flags & StringTokenFlags.Format) !== 0;
         let isInNamedUnicodeEscape = false;
-        let escapedValueParts: number[] = [];
+        const start = this._cs.position;
+        let escapedValueLength = 0;
+        const getEscapedValue = () => this._cs.getText().slice(start, start + escapedValueLength);
```

Comment:
This is where the real change happens. Instead of building the escaped value from the the char codes, it can just be built from the literal string itself.